Vagrantfile

```diff
@@ -9,6 +9,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     config.vm.network :forwarded_port, guest: 8282, host: 8282
     config.vm.network :forwarded_port, guest: 80, host: 8000
     config.vm.network :forwarded_port, guest: 4443, host: 4443
+    config.vm.network :forwarded_port, guest: 777, host: 777
```

Comment:
What's going on here is that we need a way for Nginx to use two different kind of certificates, 1) normal server-based SSL with validated cert signed by an external CA
2) for the API: SSL using certs signed by our own CA

There's (as far as I could tell) two solutions when you need separate configs like this, one is called SNI (Server Name Indication), and the other is with virtualhosts either using two different IPs, or two different ports.

I chose to go with using two different ports.  So if you connect on port 777, nginx forces client authentication, only allowing clients to connect if they have a cert signed with our own CA certificate.

This means from here on out, nginx configs are going to listen on 3 ports:
- 80 - for HTTP (immediate 301 redirect to 443)
- 443 - for normal HTTPS to /uber
- 777 - ONLY for client certificate authentication to /jsonrpc

It seemed like the right choice considering on-site we do a bunch of weird port forwarding and DNS voodoo which might make it harder to get the names to all match/etc.